File: PHP/Medium/Sorting/2780__Minimum_Index_of_a_Valid_Split.php
<?php
function minimumIndex($nums)
{
    $n = count($nums);
    $freq = array_count_values($nums);
    arsort($freq);
    $dominant = array_key_first($freq);

    $countLeft = 0;
    $countTotal = $freq[$dominant];

    for ($i = 0; $i < $n - 1; $i++) {
        if ($nums[$i] == $dominant) {
            $countLeft++;
        }
        if ($countLeft * 2 > ($i + 1) && ($countTotal - $countLeft) * 2 > ($n - $i - 1)) {
            return $i;
        }
    }
    return -1;
}

$nums1 = [1, 2, 2, 2];

echo minimumIndex($nums1);

File: Python/Medium/Sorting/2780__Minimum_Index_of_a_Valid_Split.py
from collections import Counter


def minimumIndex(nums: list[int]) -> int:
    n = len(nums)
    freq = Counter(nums)
    dominant = max(freq, key=freq.get)

    count_left = 0
    count_total = freq[dominant]

    for i in range(n - 1):
        if nums[i] == dominant:
            count_left += 1
        if count_left * 2 > (i + 1) and (count_total - count_left) * 2 > (n - i - 1):
            return i
    return -1


nums = [1, 2, 2, 2]
print(minimumIndex(nums))
